v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstogcs/BigtableChangeStreamsToGcsIT.java

```diff
@@ -0,0 +1,539 @@
+
+    pipelineLauncher.cancelJob(launchInfo.projectId(), launchInfo.region(), launchInfo.jobId());
+
+    Awaitility.await("Waiting for previous job to die.")
+        .atMost(Duration.ofMinutes(10))
+        .pollInterval(Duration.ofMillis(5000))
+        .until(
+            () ->
+                pipelineLauncher.getJobStatus(
+                        TestProperties.project(), TestProperties.region(), launchInfo.jobId())
+                    == JobState.CANCELLED);
+
+    // Scanning the dir one more time to capture the latest commit time
+    var predicate = new LookForBigtableRowAvroRecord(expected);
+    validateGCSContents(launchInfo, "output", predicate);
+
+    bigtableResourceManager.write(rowMutation);
+
+    final LaunchInfo launchInfo2 =
+        launchTemplate(
+            LaunchConfig.builder(testName, specPath)
+                .addParameter("bigtableChangeStreamName", persistentName)
+                .addParameter("bigtableChangeStreamResume", "true")
+                .addParameter("bigtableReadTableId", srcTable)
+                .addParameter("bigtableReadInstanceId", bigtableResourceManager.getInstanceId())
+                .addParameter("bigtableChangeStreamAppProfile", appProfileId)
+                .addParameter("outputFileFormat", "AVRO")
+                .addParameter(
+                    "bigtableChangeStreamStartTimestamp",
+                    formatTimeMicros(predicate.getEarliestCommitTime() - 1000000))
+                .addParameter("windowDuration", "10s")
+                .addParameter("gcsOutputDirectory", outputPath2)
+                .addParameter("schemaOutputFormat", "BIGTABLE_ROW"));
+
+    assertThatPipeline(launchInfo2).isRunning();
+
+    var latestPredicate = new LookForBigtableRowAvroRecord(expected);
+    validateGCSContents(launchInfo2, "output2", latestPredicate);
+    Assert.assertTrue(latestPredicate.getEarliestCommitTime() > predicate.getLatestCommitTime());
+  }
+
+  private String generateClusterName() {
+    return "teleport-c1";
+  }
+
+  private String generateTableName() {
+    return "table" + System.nanoTime();
+  }
+
+  @Test
+  public void testSingleMutationBigtableRowTextE2E() throws Exception {
+    String srcTable = generateTableName();
+
+    try {
+      launchTemplate(
+          LaunchConfig.builder(testName, specPath)
+              .addParameter("bigtableReadTableId", srcTable)
+              .addParameter("bigtableReadInstanceId", bigtableResourceManager.getInstanceId())
+              .addParameter("bigtableChangeStreamAppProfile", "anything")
+              .addParameter("outputFileFormat", "TEXT")
+              .addParameter("windowDuration", "10s")
+              .addParameter("gcsOutputDirectory", outputPath)
+              .addParameter("schemaOutputFormat", "BIGTABLE_ROW"));
+    } catch (RuntimeException e) {
+      Assert.assertTrue(e.getMessage().contains("The job failed before launch"));
```

Comment:
nit: assert* is often imported statically (not a blocker)

Reply:
I suggest we live with it for now.